src/Components/test/testassets/BasicTestApp/FocusEventComponent.razor

```diff
@@ -12,40 +12,59 @@
     <button @onclick="Clear">Clear</button>
 </p>
 
+<p>
+    A button that disappears when clicked:
+    @if (showButtonThatDisappearsWhenClicked)
+    {
+        <button id="button-that-disappears" @onfocusout="DisappearingButtonFocusOut" @onclick="MakeButtonDisappear">
+            Click me
+        </button>
+    }
+
+    Received focus out: <strong id="button-received-focus-out">@buttonReceivedFocusOut</strong>
+</p>
+
 <p>
     Another input (to distract you) <input id="other" />
 </p>
 
 @code {
-
+    bool showButtonThatDisappearsWhenClicked = true;
+    bool buttonReceivedFocusOut;
     string message;
 
     void OnFocus(FocusEventArgs e)
     {
         message += "onfocus,";
-        StateHasChanged();
```

Comment:
Removing these `StateHasChanged` calls is a bit of unrelated cleanup. They're just not needed.